cookbooks/2d_annulus_visualization/doc/2d_annulus_visualization.md

```diff
@@ -0,0 +1,115 @@
+
+    $ python pyvista_2d_annulus_example.py
+
+
+## Modifying inputs to the scripts
+
+Most of the script can be run without modification for your models. The only inputs you will need to adjust yourself are:
+
+    dir = 'String/Path/To/Your/Model/Directory/'
+
+where the model directory is the one which contains the [output/] directory.  
+
+The initial and final integer number of solution outputs you would like to visualize:
+
+    out_n_min=0 # The initial solution file to visualize: i.e. 'solution-00000.pvtu'
+    out_n_max=25 # The final solution file to visualize: i.e. 'solution-00025.pvtu'
+
+and the inner and outer radius of the annulus from the geometry of your model:
+    r_inner = 3481000 # units in m
+    r_outer = 6371000 # units in m.
+
+## What the script does
+
+This script can be treated a template and modified based on the fields you are interested in examining. A simple pyvista plot to show and not save the image has the basic form:
+
+    pl = pv.Plotter()
+    pl.add_mesh(
+        mesh,
+        scalars="p"
+        clim=[0,1e11], # color bar range from 0 to 100 GPa
+        cmap="RdBu",   # color map to use
+        lighting=False, # lighting is useful mainly to show shadows on 3D meshes
+                        # but we are plotting in 2D
+    )
+    pl.view_xy()
+    pl.show()
+    pl.clear()
+    pl.close()
+
+Adding the mesh titled "mesh" and informing the plotter which scalar value to plot on that mesh(in this case the pressure field which is labeled "p") is the minimum you need to plot using pyvista.
+
+In the script we plot the field "v_phi", the angular component of the velocity field, which is not an output of ASPECT. To create this new point field, the script performs several operations to:
+
+    1. Extract the x and y coordinates of the mesh points. We do this in order to do operations on those values individually since the mesh points are normally a Nxd array where N is the number of mesh points and d is the dimension. We transpose the array (now dxN), and index the first two values as the x and y coordinate arrays respectively. We save the coordinates as new point fields of the mesh called "x" and "y", which are Nx1 arrays that can be operated on and with other Nx1 arrays such temperature, and they can be plotted as a scalar field.
+    2. Extract the x and y components of the velocity vectors at every point on the mesh in a similar way as extracting x and y coordinates.
+    3. We can use the new point arrays "x", "y", "v_x" and "v_y", which are Nx1 numpy array objects, to compute the components of the velocity in polar and radial coordinates and plot those fields.
+   
+The radial component of the velocity field from cartesian coordinates is
+```{math}
+v_r = \frac{x\dot{x}+y\dot{y}}{\sqrt{x^2+y^2}},
```

Comment:
may be add afterwards where \dot{x} is the x-component of the velocity. etc ... ?